lib/datasource.js

```diff
@@ -632,6 +632,7 @@ DataSource.prototype.setupDataAccess = function(modelClass, settings) {
       if (idProp.generated && forceId !== false) {
         forceId = true;
       }
+      settings.forceId = forceId;
```

Comment:
I find this very suspicious, please let me try to investigate this myself to understand better what's going on, before we move forward.